public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -16,6 +16,66 @@ export function alignRange(start, end, step) {
   };
 }
 
+const keywords = 'by|without|on|ignoring|group_left|group_right';
+const builtInWords = [
+  keywords,
+  'count|count_values|min|max|avg|sum|stddev|stdvar|bottomk|topk|quantile',
+  'true|false|null|__name__|job',
+  'abs|absent|ceil|changes|clamp_max|clamp_min|count_scalar|day_of_month|day_of_week|days_in_month|delta|deriv',
+  'drop_common_labels|exp|floor|histogram_quantile|holt_winters|hour|idelta|increase|irate|label_replace|ln|log2',
+  'log10|minute|month|predict_linear|rate|resets|round|scalar|sort|sort_desc|sqrt|time|vector|year|avg_over_time',
+  'min_over_time|max_over_time|sum_over_time|count_over_time|quantile_over_time|stddev_over_time|stdvar_over_time',
+]
+  .join('|')
+  .split('|');
+
+// addLabelToQuery('foo', 'bar', 'baz') => 'foo{bar="baz"}'
+export function addLabelToQuery(query: string, key: string, value: string): string {
+  if (!key || !value) {
+    throw new Error('Need label to add to query.');
+  }
+
+  // Add empty selector to bare metric name
+  let previousWord;
+  query = query.replace(/(\w+)\b(?![\({=",])/g, (match, word) => {
```

Comment:
This works wrong in some cases. Try to use test case like this:
```js
expect(addLabelToQuery('foo{instance="my-host.com:9100"}', 'bar', 'baz')).toBe('foo{instance="my-host.com:9100",bar="baz"}');
```
This produces `foo{instance="my{bar="baz"}-host{bar="baz"}.com{bar="baz"}`, because `query.replace(...` replaces `foo{instance="my-host.com:9100"}` by `foo{instance="my{}-host{}.com{}:9100"}`